### ISSUE TITLE
Incompatible with faust 

### ISSUE THREAD

**Issue by adithep** [microsoft/pyright#2128]
Note: if you are reporting a wrong signature of a function or a class in the standard library, then the typeshed tracker is better suited for this report: https://github.com/python/typeshed/issues.

**Describe the bug**
Does not work with faust library.

**To Reproduce**
Create faust project and use any decorator function, for example:

```python
@app.task()
async def hello():
    pass
```

**Expected behavior**
Correctly typed, no errors.

**Screenshots or Code**

```
[Pyright reportGeneralTypeIssues] [E] Argument of type "() -> Coroutine[Any, Any, None]" cannot be assigned to parameter of type "AppT"
¬†¬†"object" is incompatible with "AppT"
```

**VS Code extension or command-line**
coc-pyright for vim

I understand if this is more of a faust issue than pyright: If this is the case, would appreciate some work arround.

**Comment by erictraut** [comment]
I don't know anything about the faust library. I presume it's installed by `pip install faust`? Could you provide a more complete example? How are you importing `app`?

It appears that faust is not a typed library. Where are you getting the type information? Have you downloaded and installed type stubs for the library? Do you have the "useLibraryCodeForTypes" setting enabled? By default, it's disabled in pyright. I don't recommend enabling this option if you are enabling type checking because the types inferred from library code is often incomplete and inaccurate. This option was added primarily for users who were not interested in type checking but still wanted basic completion suggestions even if they were not always complete.

**Comment by adithep** [comment]
This is my own stupidity

Change from 

```python
@app.task()
async def hello():
    pass
```

to

```python
@app.task
async def hello():
    pass
```

Works. sorry to have wasted your time. Btw faust is a typed library.

https://github.com/robinhood/faust
